### PR TITLE
Integrate GH actions (DO NOT MERGE)

### PR DESCRIPTION
**Pull request by LaurenceWarne**
Hi again, small PR for #191, the main changes are adding an `eldev` file (to call `eldev -dtT -C compile --warnings-as-errors` in CI) and the GH actions file using https://github.com/purcell/setup-emacs.

Could look into `cask` or even forego a build tool completely if that's your preference.  As mentioned in #191 we might want to wait until a new version of `transient` is released before merging, else/otherwise feel free to give feedback, etc :slightly_smiling_face: 

Example run: https://github.com/LaurenceWarne/docker.el/actions

Thanks!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/Eldev-local
+/.eldev
```
Comment (Silex) [inline .gitignore:2]: "New newline at end of file" I think it's better with a newline.

File: Eldev [hunk 2]
```diff
@@ -0,0 +1,4 @@
+; -*- mode: emacs-lisp; lexical-binding: t; no-byte-compile: t -*-
+
+;; Autodetermined by `eldev init'.
+(eldev-use-package-archive 'melpa-stable)
```
Comment (Silex) [inline Eldev:4]: We probably want to test stable *and* unstable no?

**Comment by Silex** [comment]
LGTM, but as said in the comments it makes sense to also test for MELPA unstable.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -32,6 +32,12 @@ jobs:
     - name: Check out the source code
       uses: actions/checkout@v2
 
-    - name: Test Byte Compilation
+    - name: Test Byte Compilation Against melpa stable
       run: |
-        eldev -dtT -C compile --warnings-as-errors
+        eldev -dtT -C --stable compile --warnings-as-errors
+
+    - name: Test Byte Compilation Against melpa unstable
```
Comment (Silex) [inline .github/workflows/test.yml:39]: Is there a rationale for the weird camelization going on here? Why not `Test byte compilation against Melpa unstable`?

Comment (LaurenceWarne) [reply .github/workflows/test.yml]: Nope :smile:

**Comment by Silex** [comment]
Thanks :+1:

**Comment by LaurenceWarne** [comment]
Another consideration would be if you wanted the different Emacs versions to be fail fast or not, I think currently it is.

**Comment by Silex** [comment]
Hum, actually I didn't catch that you simply tested sequentially.

I think we need one job for stable and one job for unstable. Right now it all fails because it's grouped together.

**Comment by LaurenceWarne** [comment]
> Hum, actually I didn't catch that you simply tested sequentially.
> 
> I think we need one job for stable and one job for unstable. Right now it all fails because it's grouped together.

Or could go for the lazp option and comment out the stable testing :smile: , sorry was not expecting you to merge :sweat_smile:

**Comment by Silex** [comment]
Feel free to comment on https://github.com/Silex/docker.el/commit/7b11b505cf8db0cf04ceb5bb23ddea65027efca8

**Comment by LaurenceWarne** [comment]
> Feel free to comment on [7b11b50](https://github.com/Silex/docker.el/commit/7b11b505cf8db0cf04ceb5bb23ddea65027efca8)

Wow that was fast, can you run the CI from that branch and see?

**Comment by Silex** [comment]
https://github.com/Silex/docker.el/actions/runs/1941593236

Just need to find out how to make it continue on errors

**Comment by LaurenceWarne** [comment]
`fail-fast: false` in `strategy` I think.

**Comment by Silex** [comment]
Thanks yes that's it: https://github.com/Silex/docker.el/actions/runs/1941606556

**Comment by LaurenceWarne** [comment]
Nice, looks good, I can create an issue if you like for the `transient` version stuff.

**Comment by Silex** [comment]
Thanks it's was a nice & easy addition!

Hum yes good idea about transient, you can mention https://github.com/Silex/docker.el/actions/runs/1941613092 as an example of test failing because of transient (I won't delete this run).

**Pull request merged**